src/tasks/TaskManager.ts

```diff
@@ -1148,6 +1148,7 @@ class TaskManager {
     this.logger.debug(`Requeued Task ${taskIdEncoded}`);
   }
 
+  @ready(new tasksErrors.ErrorTaskManagerNotRunning())
```

Comment:
This should not be needed. It is a protected function so you don't need a `ready` decorator.

Reply:
It's protected here but it's callable at any time with `task.cancel()` or `task.promise().cancel()`. I added it while debugging but it's not really needed.

Reply:
I've removed it.